### PR TITLE
Delete more java only code

### PR DESCRIPTION
**Pull request by tasomaniac**
I realized we can delete even more :)

### CODE REVIEW

**Review by martinbonnin — APPROVED** [review]
It currently fails because `Optional` is still used from `apollo-runtime`. If everything is moved to Kotlin, we can use Kotlin optionals but there might be some Java files to convert.

**Comment by tasomaniac** [comment]
Yeah, I thought this would be easier. I moved Optional to the runtime module. And converted some more unit tests to kotlin in order to remove this unnecessary abstraction over map.

File: apollo-api/build.gradle.kts [hunk 1]
```diff
@@ -20,9 +20,7 @@ kotlin {
     }
   }
 
-  jvm {
-    withJava()
```
Comment (martinbonnin) [inline apollo-api/build.gradle.kts:24]: Nice :)

**Review by martinbonnin — APPROVED** [review]

**Review by sav007 — APPROVED** [review]
holly .... -1585 :)

**Pull request merged**